BeerProject/src/test/java/wawer/kamil/beerproject/controllers/UserControllerTest.java

```diff
@@ -1,125 +1,248 @@
+        assertThat(newUserResponseEntity.getBody().getEmail()).isEqualTo("test@email.com");
+        assertThat(newUserResponseEntity.getBody().isAccountNonExpired()).isTrue();
+        assertThat(newUserResponseEntity.getBody().isAccountNonLocked()).isTrue();
+        assertThat(newUserResponseEntity.getBody().isCredentialsNonExpired()).isTrue();
+        assertThat(newUserResponseEntity.getBody().isEnabled()).isTrue();
     }
 
     @Test
-    public void should_return_saved_user_which_equals_to_user_updated_by_controller() throws NoContentException {
-        when(mapper.map(userDTO, User.class)).thenReturn(user);
-        when(service.updateUser(ID, user)).thenReturn(user);
-        when(mapper.map(user, UserDTO.class)).thenReturn(userDTO);
-        when(user.getId()).thenReturn(ID);
-        assertEquals(ResponseEntity.status(HttpStatus.OK).body(userDTO),controller.updateUser(ID, userDTO));
+    @DisplayName("Test - should throw exception when user not found during update")
+    public void should_throw_exception_when_user_not_found_during_update() throws NoContentException {
+        //given
+        when(userService.updateUser(USER_ID, userRequest)).thenThrow(NoContentException.class);
+
+        //when
+        assertThatThrownBy(() -> controller.updateUser(USER_ID, userRequest));
     }
 
     @Test
-    public void should_return_status_code_which_equals_to_controller_status_code_base_on_update_user() throws NoContentException {
-        when(mapper.map(userDTO, User.class)).thenReturn(user);
-        when(service.updateUser(ID, user)).thenReturn(user);
-        when(mapper.map(user, UserDTO.class)).thenReturn(userDTO);
-        assertEquals(HttpStatus.OK,controller.updateUser(ID, userDTO).getStatusCode());
+    @DisplayName("Test - should return response entity when deleting user permanently")
+    public void should_return_response_entity_when_deleting_user_permanently() throws NoContentException {
+        //when
+        ResponseEntity<?> responseEntity = controller.deleteUserPermanently(USER_ID);
+
+        //than
+        assertThat(responseEntity).isNotNull();
+        assertThat(responseEntity.getStatusCode()).isEqualTo(HttpStatus.NO_CONTENT);
+        assertThat(responseEntity.getBody()).isNull();
     }
 
     @Test
-    public void should_return_status_code_which_equals_to_controller_status_code_base_on_deleted_user() throws NoContentException {
-        assertEquals(HttpStatus.NO_CONTENT, controller.deleteUser(ID).getStatusCode());
+    @DisplayName("Test - should throw exception when user not found during deleting user permanently")
+    public void should_throw_exception_when_user_not_found_during_deleting_user_permanently() throws NoContentException {
+        //given
+        when(controller.deleteUserPermanently(USER_ID)).thenThrow(NoContentException.class);
+
+        //when
+        assertThatThrownBy(() -> controller.deleteUserPermanently(USER_ID));
     }
```

Comment:
może tam gdzie assert to zrobić //then bo assert to już koniec testu, i zamiast //given to //when

Reply:
jw

Reply:
jw